lib/utils.js

```diff
@@ -392,6 +392,29 @@ var Utils = module.exports = {
 
   validateParameter: function(value, expectation, options) {
     return ParameterValidator.check(value, expectation, options);
+  },
+
+  formatReferences: function (obj) {
+    var references = {};
+
+    if (!obj || !_.isPlainObject(obj) || !obj.references) {
+      return obj;
+    }
+
+    if (_.isPlainObject(obj.references)) {
+      references = obj.references;
+    } else {
+      deprecate('Non-object references property. Expected { references: { model: "value" } } instead of { references: "value" }.');
+      references.model = obj.references;
```

Comment:
Please insert `references.key = obj.referencesKey;` here. That way you do not need the next if case `if (obj.referencesKey) {}`.
If I'm not mistaken there is no use case of the referencesKey allone, so you do not have to check for that.